src/Umbraco.Infrastructure/Persistence/Repositories/Implement/DistributedJobRepository.cs

```
@@ -0,0 +1,124 @@
+﻿using NPoco;
+using Umbraco.Cms.Core.Exceptions;
+using Umbraco.Cms.Infrastructure.BackgroundJobs;
+using Umbraco.Cms.Infrastructure.Models;
+using Umbraco.Cms.Infrastructure.Persistence.Dtos;
+using Umbraco.Cms.Infrastructure.Scoping;
+using Umbraco.Extensions;
+
+namespace Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement;
+
+/// <inheritdoc />
+internal class DistributedJobRepository(IScopeAccessor scopeAccessor) : IDistributedJobRepository
+{
+    /// <inheritdoc />
+    public string? GetRunnable()
+    {
+        if (scopeAccessor.AmbientScope is null)
+        {
+            return null;
+        }
```

I think we should throw an exception if there's no scope :smile: